src/main/scala/io/iohk/ethereum/faucet/jsonrpc/FaucetRpcService.scala

```diff
@@ -1,56 +1,59 @@
 package io.iohk.ethereum.faucet.jsonrpc
 
-import akka.util.ByteString
-import io.iohk.ethereum.domain.{Address, Transaction}
-import io.iohk.ethereum.faucet.FaucetConfig
-import io.iohk.ethereum.faucet.FaucetStatus.FaucetUnavailable
+import akka.actor.{ActorRef, ActorSystem}
+import akka.pattern.RetrySupport
+import akka.util.Timeout
+import io.iohk.ethereum.faucet.{FaucetConfig, FaucetHandler, FaucetSupervisor}
+import io.iohk.ethereum.faucet.FaucetHandler.{FaucetHandlerMsg, FaucetHandlerResponse}
+import io.iohk.ethereum.jsonrpc.AkkaTaskOps._
 import io.iohk.ethereum.faucet.jsonrpc.FaucetDomain.{SendFundsRequest, SendFundsResponse, StatusRequest, StatusResponse}
 import io.iohk.ethereum.jsonrpc.{JsonRpcError, ServiceResponse}
-import io.iohk.ethereum.keystore.KeyStore
-import io.iohk.ethereum.mallet.service.RpcClient
-import io.iohk.ethereum.network.p2p.messages.CommonMessages.SignedTransactions.SignedTransactionEnc
-import io.iohk.ethereum.rlp
-import io.iohk.ethereum.utils.{ByteStringUtils, Logger}
+import io.iohk.ethereum.utils.Logger
 import monix.eval.Task
 
-class FaucetRpcService(rpcClient: RpcClient, keyStore: KeyStore, config: FaucetConfig) extends Logger {
+import scala.concurrent.ExecutionContext
 
-  private val wallet = keyStore.unlockAccount(config.walletAddress, config.walletPassword) match {
-    case Right(w) =>
-      log.info(s"wallet unlocked for use in faucet (${config.walletAddress})")
-      w
-    case Left(err) =>
-      throw new RuntimeException(s"Cannot unlock wallet for use in faucet (${config.walletAddress}), because of $err")
-  }
+class FaucetRpcService(config: FaucetConfig)(implicit system: ActorSystem) extends RetrySupport with Logger {
```

Comment:
// TODO: Add unit tests for this class. (it could be done in a different task)

Reply:
Add task: ETCM-395